### PR TITLE
CA-8430: add-tests-for dt-webhook

### PR DESCRIPTION
**Pull request by dietmar-91**
This PR includes:

- initial test config
- initial unit/integration tests for dt_webhook.py

### CODE REVIEW

**Review by agrimmer — DISMISSED** [review]
LGTM

File: tests/integration/test_dt_webhook.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+url = f'http://{args["host"]}:{args["port"]}/event'
+payload = json.dumps({"eventId": "1A2B3C"})
+headers = {"Authorization": "Bearer " + args["token"]}
+
+async def run_webhook() -> None:  # noqa: FA102, D103
+    """Start webhook."""
+    await dt_webhook(asyncio.Queue(), args)
+
+@pytest.mark.asyncio
+async def test_with_incorrect_path():
```
Comment (christian-kreuzberger-dtx) [inline tests/integration/test_dt_webhook.py:27]: can you please add a comment, specifying what the test is doing?
e.g.:
```python
// When posting data to an incorrect path, HTTP 404 Not Found should be returned
```

File: tests/integration/test_dt_webhook.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    async def do_request():
+        async with aiohttp.ClientSession(headers=headers) as session:
+            async with session.post(f'http://{args["host"]}:{args["port"]}/something', data=payload) as resp:
+                plugin_task.cancel()
+                assert resp.status == HTTPStatus.NOT_FOUND
```
Comment (christian-kreuzberger-dtx) [inline tests/integration/test_dt_webhook.py:32]: I don't know how tests with async io work, but I would expect the `assert` statement to come somewhere after the `await asyncio.gather` - though that would require `resp` to be passed to the outside.
Long story short, I think this is fine :D

File: tests/integration/test_dt_webhook.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+async def run_webhook() -> None:  # noqa: FA102, D103
+    """Start webhook."""
+    await dt_webhook(asyncio.Queue(), args)
+
+@pytest.mark.asyncio
+async def test_with_incorrect_path():
+    async def do_request():
+        async with aiohttp.ClientSession(headers=headers) as session:
+            async with session.post(f'http://{args["host"]}:{args["port"]}/something', data=payload) as resp:
+                plugin_task.cancel()
```
Comment (christian-kreuzberger-dtx) [inline tests/integration/test_dt_webhook.py:31]: can you comment and describe what `plugin_task.cancel()` is needed for?

File: tests/unit/test_dt_webhook.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+)
+
+args = {
+    "host": "127.0.0.1",
+    "port": 1234,
+    "token": "thisisnotanactualtoken",
+}
+
+url = f'http://{args["host"]}:{args["port"]}/event'
+payload = {"eventId": "1A2B3C"}
```
Comment (christian-kreuzberger-dtx) [inline tests/unit/test_dt_webhook.py:18]: is `payload` actually needed for unit tests? Or did you miss a unit test where it should be used?

File: tests/unit/test_dt_webhook.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    _set_app_attributes,
+)
+
+args = {
+    "host": "127.0.0.1",
+    "port": 1234,
+    "token": "thisisnotanactualtoken",
+}
+
+url = f'http://{args["host"]}:{args["port"]}/event'
```
Comment (christian-kreuzberger-dtx) [inline tests/unit/test_dt_webhook.py:17]: is `url` actually needed for unit test? Or did you miss a unit test where it should be used?

File: tests/unit/test_dt_webhook.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+payload = {"eventId": "1A2B3C"}
+headers = {"Authorization": "Bearer " + args["token"]}
+
+
+def test_parse_token_with_incorrect_token():
+    with pytest.raises(web.HTTPUnauthorized,
+    match="Invalid authorization token"):
```
Comment (christian-kreuzberger-dtx) [inline tests/unit/test_dt_webhook.py:24]: Can this be moved into a single line instead?
```suggestion
    with pytest.raises(web.HTTPUnauthorized, match="Invalid authorization token"):
```

File: tests/unit/test_dt_webhook.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+def test_set_app_attributes():
+    app_attrs = _set_app_attributes(args)
+    assert app_attrs["host"] == "127.0.0.1"
+    assert app_attrs["port"] == 1234
+    assert app_attrs["token"] == "thisisnotanactualtoken"
+
+
+def test_set_app_attributes_without_port():
+    with pytest.raises(ValueError, match="Port is missing as an argument" ):
```
Comment (christian-kreuzberger-dtx) [inline tests/unit/test_dt_webhook.py:41]: Is there a linter configuration that forces us to have a whitespace after the string? If not, I would prefer to remove it.
```suggestion
    with pytest.raises(ValueError, match="Port is missing as an argument"):
```

File: tests/unit/test_dt_webhook.py (hunk 5 above)
Comment (dietmar-91) [reply tests/unit/test_dt_webhook.py:17]: No good catch. I will remove it.

File: tests/unit/test_dt_webhook.py (hunk 4 above)
Comment (dietmar-91) [reply tests/unit/test_dt_webhook.py:18]: I will also remove it.

File: tests/unit/test_dt_webhook.py (hunk 7 above)
Comment (dietmar-91) [reply tests/unit/test_dt_webhook.py:41]: No should be not linter config. I will remove it.

File: tests/integration/test_dt_webhook.py (hunk 3 above)
Comment (dietmar-91) [reply tests/integration/test_dt_webhook.py:31]: Without canceling the task the plugin coroutine which is wrapped into the task is never stopped and also the test is never terminated.

File: tests/integration/test_dt_webhook.py (hunk 1 above)
Comment (dietmar-91) [reply tests/integration/test_dt_webhook.py:27]: I added docstrings to the tests.

File: tests/integration/test_dt_webhook.py
Comment (christian-kreuzberger-dtx) [inline tests/integration/test_dt_webhook.py:1]: Please add to the docs:
* how to run integration tests
* missing step to install `pytest-asyncio`

Comment (dietmar-91) [reply tests/integration/test_dt_webhook.py:1]: Done. I added the missing parts from your comment and a section for `tox `and `linting`.

**Review by christian-kreuzberger-dtx — APPROVED** [review]

**Review by TannerGabriel — APPROVED** [review]

**Pull request merged**